src/caching.ts

```diff
@@ -114,9 +115,15 @@ export function createCache<S extends Store, C extends Config>(
           const cacheTTL = typeof ttl === 'function' ? ttl(value) : ttl;
           const remainingTtl = await store.ttl(key);
           if (remainingTtl !== -1 && remainingTtl < refreshThresholdConfig) {
-            coalesceAsync(`+++${key}`, fn).then((result) =>
-              store.set<T>(key, result, cacheTTL),
-            );
+            coalesceAsync(`+++${key}`, fn)
+              .then((result) => store.set<T>(key, result, cacheTTL))
+              .catch((error) => {
+                if (args?.onBackgroundRefreshError) {
+                  args.onBackgroundRefreshError(error);
+                } else {
+                  return Promise.reject(error);
```

Comment:
This should behave as before if `onBackgroundRefreshError` is not set. However, should this error be anyway wrapped to some custom error (`BackgroundRefreshError`...?) to make it easier to figure out where the unhandled promise is originating from?